projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php

```diff
@@ -148,31 +147,142 @@ function wpcom_add_hosting_menu() {
+		'manage_options',
+		$activity_log_url,
+		null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
+	);
+
 	add_submenu_page(
 		'jetpack',
+		__( 'VaultPress', 'jetpack-mu-wpcom' ),
+		__( 'VaultPress', 'jetpack-mu-wpcom' ),
+		'manage_options',
+		$vaultpress_url,
+		null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
+	);
+
+	add_submenu_page(
+		'jetpack',
+		__( 'Monetize', 'jetpack-mu-wpcom' ),
 		__( 'Monetize', 'jetpack-mu-wpcom' ),
-		__( 'Monetize', 'jetpack-mu-wpcom' ) . ' <span class="dashicons dashicons-external"></span>',
 		'manage_options',
-		esc_url( Redirect::get_url( 'calypso-monetize' ) ),
-		null // @phan-suppress-current-line PhanTypeMismatchArgumentProbablyReal
+		$monetize_url,
+		null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
 	);
+
+	if ( $is_atomic_site ) {
+		add_submenu_page(
+			'jetpack',
+			__( 'Scan', 'jetpack-mu-wpcom' ),
+			__( 'Scan', 'jetpack-mu-wpcom' ),
+			'manage_options',
+			$scan_url,
+			null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
+		);
+	}
+
+	add_submenu_page(
+		'jetpack',
+		__( 'Subscribers', 'jetpack-mu-wpcom' ),
+		__( 'Subscribers', 'jetpack-mu-wpcom' ),
+		'manage_options',
+		$subscribers_url,
+		null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
+	);
+
+	if ( $is_simple_site ) {
+		add_submenu_page(
+			'jetpack',
+			__( 'Newsletter', 'jetpack-mu-wpcom' ),
+			__( 'Newsletter', 'jetpack-mu-wpcom' ),
+			'manage_options',
+			$newsletter_url,
+			null // @phan-suppress-next-line PhanTypeMismatchArgumentProbablyReal -- Core should ideally document null for no-callback arg. https://core.trac.wordpress.org/ticket/52539.
+		);
+	}
+
+	// Re-order menu.
+	global $submenu;
+	if ( ! isset( $submenu['jetpack'] ) ) {
+		return;
+	}
+
+	$desired_order   = array(
+		'my-jetpack',
+		'stats',
+		$activity_log_url,
+		$vaultpress_url,
+		'akismet-key-config',
+		'jetpack-search',
+		$scan_url,
+		$monetize_url,
+		$subscribers_url,
+	);
+	$ordered_submenu = array();
+
+	// Re-add submenu items in the desired order.
+	foreach ( $desired_order as $slug ) {
+		foreach ( $submenu['jetpack'] as $item ) {
+			if ( $item[2] === $slug ) {
+				$ordered_submenu[] = $item;
+			}
+		}
+	}
+
+	// Add any remaining submenu items.
+	foreach ( $submenu['jetpack'] as $item ) {
+		if ( ! in_array( $item[2], $desired_order, true ) ) {
+			$ordered_submenu[] = $item;
+		}
+	}
```

Comment:
Can't this be simplified to ( O(n) vs. O(n*m) + O(m) = O(n*m) ):
```
$ordered = array(
	'my-jetpack' => null,
	'stats' => null,
	$activity_log_url => null,
	$vaultpress_url => null,
	'akismet-key-config' => null,
	'jetpack-search' => null,
	$scan_url => null,
	$monetize_url => null,
	$subscribers_url => null,
);

// Assign values to the ordered array, if we do not have a specific order we append it to the end.
foreach ( $submenu['jetpack'] as $item ) {
	if ( array_key_exists( $item[2], $ordered ) ) {
		$ordered[ $item[2] ] = $item;
	} else {
		$ordered[] = $item; // Add remaining items at the end
	}
}

$ordered = array_filter( $ordered );


```

Reply:
That would surely decrease the run time, but I'd argue it increases the readability complexity 😅 

So considering this is a small array, I think having a higher O(n) complexity is an acceptable trade-off.